oleaut32.go

```
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
 }
 
+func (v *VARIANT) LVal() int32 {
+	value, err := v.GetLVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetLVal() (int32, error) {
+	if v.Vt != VT_I4 {
+		return 0, fmt.Errorf("Error: LVal() v.Vt !=  VT_I4, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_I4)(unsafe.Pointer(v))
+	return p.lVal, nil
+}
+
+func (v *VARIANT) SetLVal(value int32) {
+	v.Vt = VT_I4
+	p := (*VAR_I4)(unsafe.Pointer(v))
+	p.lVal = value
+}
+
+func (v *VARIANT) ULVal() uint32 {
+	value, err := v.GetULVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetULVal() (uint32, error) {
+	if v.Vt != VT_UI4 {
+		return 0, fmt.Errorf("Error: LVal() v.Vt !=  VT_UI4, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_UI4)(unsafe.Pointer(v))
+	return p.ulVal, nil
+}
+
+func (v *VARIANT) SetULVal(value uint32) {
+	v.Vt = VT_UI4
+	p := (*VAR_UI4)(unsafe.Pointer(v))
+	p.ulVal = value
+}
+
+func (v *VARIANT) BoolVal() VARIANT_BOOL {
+	value, err := v.GetBoolVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetBoolVal() (VARIANT_BOOL, error) {
+	if v.Vt != VT_BOOL {
+		return VARIANT_FALSE, fmt.Errorf("Error: BoolVal() v.Vt !=  VT_BOOL, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_BOOL)(unsafe.Pointer(v))
+	return p.boolVal, nil
+}
+
+func (v *VARIANT) SetBoolVal(value VARIANT_BOOL) {
+	v.Vt = VT_BOOL
+	p := (*VAR_BOOL)(unsafe.Pointer(v))
+	p.boolVal = value
+}
+
+func (v *VARIANT) BstrVal() *uint16 {
+	value, err := v.GetBstrVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetBstrVal() (*uint16, error) {
+	if v.Vt != VT_BSTR {
+		return nil, fmt.Errorf("Error: BstrVal() v.Vt !=  VT_BSTR, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_BSTR)(unsafe.Pointer(v))
+	return p.bstrVal, nil
+}
+
+func (v *VARIANT) SetBstrVal(value *uint16) {
+	v.Vt = VT_BSTR
+	p := (*VAR_BSTR)(unsafe.Pointer(v))
+	p.bstrVal = value
+}
+
+func (v *VARIANT) PDispVal() *IDispatch {
```

This should be called `MustPDispatch` and use `panic` instead of `log.Fatal` to allow recovery.